src/WebJobs.Script/Description/Rpc/WorkerFunctionDescriptorProvider.cs

```diff
@@ -40,5 +43,27 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
             });
             return new WorkerLanguageInvoker(Host, triggerMetadata, functionMetadata, inputBindings, outputBindings, inputBuffer);
         }
+
+        protected override Collection<ParameterDescriptor> GetFunctionParameters(IFunctionInvoker functionInvoker, FunctionMetadata functionMetadata,
+            BindingMetadata triggerMetadata, Collection<CustomAttributeBuilder> methodAttributes, Collection<FunctionBinding> inputBindings, Collection<FunctionBinding> outputBindings)
+        {
+            var parameters = base.GetFunctionParameters(functionInvoker, functionMetadata, triggerMetadata, methodAttributes, inputBindings, outputBindings);
+
+            var bindings = inputBindings.Union(outputBindings);
+
+            var returnsSomething = bindings.SingleOrDefault(b => b.Metadata.ReturnsSomething);
+            if (returnsSomething != null)
+            {
+                Type type = typeof(Task<object>);
+                var type2 = type.GetGenericArguments()[0];
+                var byRefType = type2.MakeByRefType();
+
+                ParameterDescriptor returnDescriptor = new ParameterDescriptor(ScriptConstants.SystemReturnParameterName, byRefType);
```

Comment:
>SystemReturnParameterName [](start = 95, length = 25)

Is the idea that the Function IL generator detects this new parameter and that causes it to generate a real C# return value? And then the WebJobs SDK indexer picks that up and creates a binding to the return value? And then the return value from the invoker will get passed to that?

Reply:
This is a good question. I think there's some missing context and it would good to share a high level proposal/design for the feature to make sure the goals are clear.